crates/typst-cli/src/update.rs

```diff
@@ -40,7 +40,13 @@ pub fn update(command: &UpdateCommand) -> StrResult<()> {
         }
     }
 
-    let backup_path = backup_path()?;
+    let backup_dir = command.backup_path.clone().map(Ok).unwrap_or_else(backup_path)?;
+
+    fs::create_dir_all(&backup_dir)
+        .map_err(|err| eco_format!("failed to create backup directory ({err})"))?;
+
+    let backup_path = backup_dir.join("typst_backup.part");
```

Comment:
I wonder, if the user specifies `--backup-path`, shouldn't that be the full path rather than a directory to which `typst_backup.part` is joined?

Reply:
I think this is unnecessary in principle. Also adds a bit of a burden to remember the correct filename when using `--revert`.

Reply:
I was confused mostly by the variable naming here. It points towards an inconsistency, either in the CLI arg naming or in the source code.

Reply:
Fair. I think my argument also doesn't really hold because, in practice, you'd have to remember the directory regardless.

I considered that keeping it as a directory would probably be more "future-proof" since we can add more files to the backup without problems, but I guess we could just generate a `tar` instead then.

I thought about renaming it to  `--backup-file-path` or `--backup-file` to make the difference from other parameters more apparent, but wasn't sure. Maybe changing it to a file isn't that bad anyway?

Reply:
Alright, I've made the following changes:

1. Renamed `--packages-path` to `--package-path`.
2. Changed `--backup-path` to point to the backup file and updated the relevant arg docs.

I think this should be ok now.